File: app/src/main/java/com/moictab/curriculumvitae/controller/ExperienceController.java
package com.moictab.curriculumvitae.controller;

import android.content.Context;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.moictab.curriculumvitae.model.Experience;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Created by moict on 15/10/2017.
 */

public class ExperienceController extends BaseController {

    private final static String EXPERIENCES_FILENAME = "experiences.json";

    public List<Experience> GetExperiencesFromAssets(Context context) {
        Type list = new TypeToken<List<Experience>>() {
        }.getType();

        return new Gson().fromJson(GetJsonFromResources(context, EXPERIENCES_FILENAME), list);
    }

    public Experience GetExperienceFromAssetsById(Context context, int id) {
        List<Experience> experiences = GetExperiencesFromAssets(context);
        return experiences.get(id);
    }
}

File: app/src/main/java/com/moictab/curriculumvitae/controller/ExperienceController.kt
package com.moictab.curriculumvitae.controller

import android.content.Context

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.moictab.curriculumvitae.model.Experience

class ExperienceController : BaseController() {
  fun getExperiencesFromAssets(context: Context): List<Experience>? {
    val list = object : TypeToken<List<Experience>>() {}.type
    return Gson().fromJson<List<Experience>>(getJsonFromResources(context, EXPERIENCES_FILENAME), list)
  }

  fun getExperienceFromAssetsById(context: Context, id: Int): Experience {
    val experiences = getExperiencesFromAssets(context)
    return experiences!![id]
  }

  companion object {
    private const val EXPERIENCES_FILENAME = "experiences.json"
  }
}
